connect/runtime/src/main/java/org/apache/kafka/connect/runtime/WorkerConnector.java

```diff
@@ -191,6 +192,21 @@ public boolean isSinkConnector() {
         return SinkConnector.class.isAssignableFrom(connector.getClass());
     }
 
+    public ConnectorType getConnectorType() {
+        return isSinkConnector() ? ConnectorType.SINK : ConnectorType.SOURCE;
+    }
+
+    public static ConnectorType getConnectorType(String clsname) {
+        if (null == clsname || clsname.isEmpty())
+            return null;
```

Comment:
Again, lets avoid `null` stuff if possible? Should we just throw an `IllegalArgumentException` here instead since we really shouldn't ever be asking for `null` or empty connector names?